### ISSUE TITLE
toJSON not deleting fields in associations

### ISSUE THREAD

**Issue by NizarBlond** [balderdashy/sails#2688]
In activity.js (some model)

        // One-Way association with User model
        createdBy: {
            model: 'user'
        },

In user.js:

        toJSON: function() {
            var obj = this.toObject();
            delete obj.password;
            delete obj.createdAt;
            delete obj.updatedAt;
            return obj;
        }

When I retrieve Activity object, 

             "createdBy": {
                 "name": "John Smith",
                 "password": "$2a$10$D6zIThptFHyzA48mOwPUPOSaO83z0Mlw6JAuXgiKUJhCuz.ZIgj1q",
                 "createdAt": "2015-02-09T19:38:48.016Z",
                 "updatedAt": "2015-02-16T18:24:55.001Z",
                 "id": "54d90cc8d9b60440148bdea3"
             },

Quick fix:
in activity.js toJSON function:

            if (obj.createdBy) {
                delete obj.createdBy.createdAt;
                delete obj.createdBy.updatedAt;
                delete obj.createdBy.password;
            }

A generic solution would be great.

**Comment by sgress454** [comment]
Good call @NizarBlond.  Thanks for the note.

**Comment by javalos** [comment]
hey @NizarBlond, @sgress454  I'll be taking a look at this today. I'll come back later if found something.

**Comment by javalos** [comment]
@NizarBlond I'm using following model definition. The `toJSON` method is filtering out correctly in Activity.js. Did you define `toJSON` within `attributes` block in User's model ?

Any other guidance in order to replicate the issue?

```js
// User.js

module.exports = {
  attributes: {
    password : {
      type : 'string'
    },

    toJSON: function() {
      var obj = this.toObject();
      delete obj.password;
      delete obj.createdAt;
      delete obj.updatedAt;
      return obj;
    }
  }
};
```

```js
// Activity.js

module.exports = {
  attributes: {
    createdBy: {
      model: 'user'
    }
  }
};
```

```js
// Result when retrieving an activity
{
    "createdBy": {
        "id": 1
    },
    "id": 1,
    "createdAt": "2015-03-24T16:18:43.220Z",
    "updatedAt": "2015-03-24T16:18:43.220Z"
}
```

**Comment by crh3675** [comment]
@NizarBlond Would it be more accurate to say that __toJSON__ is not being executed when associations are being retrieved?

**Comment by CWyrtzen** [comment]
So, is this a bug? Has it been resolved?

**Comment by sailsbot** [comment]
Thanks for posting, @NizarBlond.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
